packages/ckeditor5-dev-env/tests/translations/upload.js

```diff
@@ -10,67 +10,82 @@
 const path = require( 'path' );
 const sinon = require( 'sinon' );
 const mockery = require( 'mockery' );
-const transifexService = require( '../../lib/translations/transifex-service' );
-const fs = require( 'fs' );
 
 describe( 'upload', () => {
-	let sandbox;
+	const sandbox = sinon.sandbox.create();
 	let upload;
+	let packageNames;
+	let stubs;
 
 	beforeEach( () => {
-		sandbox = sinon.sandbox.create();
+		mockery.enable( {
+			useCleanCache: true,
+			warnOnReplace: false,
+			warnOnUnregistered: false,
+		} );
+
+		stubs = {
+			logger: {
+				info: sandbox.stub(),
+				warning: sandbox.stub(),
+				error: err => {
+					throw err;
```

Comment:
Logger only writes the logs. Logger must not throw anything. 

In your case I would do something like that (if logging the error is important):

```js
return promise()
    .catch( ( err ) => {
        log.error( err );
        throw err;
    } );
```